### DISCUSSION TITLE
Change handling of OpenBlas buffers

### DISCUSSION THREAD

**Discussion by oscardssmith** [JuliaLang/julia#44070] [Linear Algebra]
Currently, most of the ram used by julia on startup is buffers for OpenBlas. This is fine for numerical code, but wastes a bunch of memory for code that doesn't multiply matrices (I was as shocked as you are to discover that not everyone multiplies matrices). I'm not sure what the best way to achieve this is, but in Python, loading NumPy doesn't allocate a ton of ram. Given this, deferring the allocation of OpenBlas buffers until matrices are actually multiplied would increase the responsiveness of Julia for non-numeric use-cases.

**Comment by ViralBShah** [comment]
I'm making this a discussion, since there isn't a clear plan on what needs to be done here.

**Comment by PallHaraldsson** [comment]
Is it a problem? I mean is the memory allocated and then not used? Then doesn't cost physical RAM? However LinearAlgebra is being excised, and with it OpenBLAS I guess, I'm at least hoping that. BLIS.jl better anyway(?!).

**Reply by adlambert** [reply]
For Julia to go further as a general purpose language, rather than one associated with numeric stuff, then it would be useful to have the option of reducing its memory footprint. Taking chunks of RAM that are not required is not efficient and a server process which might be called on to run a lot of instances at one time would get very wasteful. 

Yes, I could go use something else that doesn't depend on LLVM etc, but I like Julia. Would like to use it in more situations like I can with P*****.

A startup option to manage the buffer allocation, select a zero if required or a smaller size?

**Reply by ViralBShah** [reply]
OpenBLAS buffers are not manageable in that way, unfortunately. However, our cold start is now 200M. And you can soon also start using `juliac` to deploy binaries for compute - if you don't need BLAS, you can avoid linking it altogether.
